### PR TITLE
feat: verify proofs for state transitions in strategies

### PR DESCRIPTION
**Pull request by pauldelucia**
Added proof verification that state transitions were included in the block and updated the chain state. After every ST, we log whether the proof was verified or not, and the error if not.

Since the state_transitions_for_block function returns a vector of StateTransition, we verify proofs with verify_state_transition_was_executed_with_proof, which is implemented for StateTransition, rather than using functions like put_to_platform_and_wait_for_response, which are implemented for Document, Identity, etc.

### CODE REVIEW

File: .env.testnet [hunk 1]
```diff
@@ -5,7 +5,7 @@ EXPLORER_DAPI_ADDRESSES=https://34.214.48.68:1443,https://35.165.50.126:1443,htt
 EXPLORER_CORE_HOST=127.0.0.1
 
 # The port number on which Dash Core RPC is listening.
-EXPLORER_CORE_RPC_PORT=19998
+EXPLORER_CORE_RPC_PORT=9998
```
Comment (QuantumExplorer) [inline .env.testnet]: This is incorrect. Testnet is 19998

File: src/config.rs [hunk 2]
```diff
@@ -47,7 +47,7 @@ impl Config {
     /// and prefixed with [LOCAL_EXPLORER_](Config::CONFIG_PREFIX).
     pub fn load() -> Self {
         // load config from .env file
-        if let Err(err) = dotenvy::from_path(".env.local") {
+        if let Err(err) = dotenvy::from_path(".env.testnet") {
```
Comment (QuantumExplorer) [inline src/config.rs]: .env

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: .env.testnet (hunk 1 above)
Comment (pauldelucia) [reply .env.testnet:8]: It only works for me if it's 9998

File: src/backend/strategies.rs [hunk 3]
```diff
@@ -922,116 +931,159 @@ pub(crate) async fn run_strategy_task<'s>(
...
+                                                    if let Some(metadata) = &v0_response.metadata {
+                                                        let actual_block_height = metadata.height;
+                                                        info!(
+                                                            "Successfully processed state transition {} ({}) for block {} (Actual block height: {})",
+                                                            index + 1, transition.name(), current_block_info.height, actual_block_height
                                                         );
+                                    
+                                                        // Verification of the proof
+                                                        if let Some(wait_for_state_transition_result_response_v0::Result::Proof(proof)) = &v0_response.result {
+                                                            // For proof verification, if it's a DocumentsBatch, include the data contract, else don't
```
Comment (QuantumExplorer) [inline src/backend/strategies.rs:1001]: Don't you want both if there is a proof and if there isn't?

Comment (QuantumExplorer) [reply src/backend/strategies.rs:1001]: Basically so we can test both out.

Comment (pauldelucia) [reply src/backend/strategies.rs:1001]: There's no functionality for that in the sdk right now so we'll have to add it there first.

**Pull request closed without merging**